### ISSUE TITLE
Allowing other archives to be pushed with --path

### ISSUE THREAD

**Issue by chibi03** [cloudfoundry-attic/cfoundry#23]
Would it be possible to allow applications other than .jar/.war/.zip files to be pushed, using --path in the cf cli? 
I'm trying to add support of .ear files in a buildpack but this keeps me from allowing them to be pushed with --path, upload_helper.rb:

``` ruby
def prepare_package(path, to)
      if path =~ /\.(jar|war|zip)$/
        CFoundry::Zip.unpack(path, to)
      elsif (war_file = Dir.glob("#{path}/*.war").first)
        CFoundry::Zip.unpack(war_file, to)
      else
        FileUtils.mkdir(to)

        exclude = UPLOAD_EXCLUDE
        if File.exists?("#{path}/.cfignore")
          exclude += File.read("#{path}/.cfignore").split(/\n+/)
        end

        files = files_to_consider(path, exclude)

        check_unreachable_links(files, path)

        copy_tree(files, path, to)

        find_sockets(to).each do |s|
          File.delete s
        end
      end
end
```

Thanks.

**Comment by ruthie** [comment]
Hi,

We are open to allowing .ear files to be pushed. Please submit a pull request if you get something working.

Thanks,

@ruthie @charliebevis

**Comment by chibi03** [comment]
Before I submit a pull request, are there any requirements for the request to get accepted? e.g. RSpec tests

**Comment by zrob** [comment]
Hi,

We do have a draft of our [Contribution Guidelines](http://bit.ly/cfcodestds) being worked on.  This should give you a good start on what we expect.

TLDR: Follow general ruby guidelines and provide full test coverage of any changes

Thanks,
@zrob & @oppegard

**Comment by zrob** [comment]
Also, we require a CLA agreement before we accept pull requests.  You can find the document [here](http://www.cloudfoundry.org/assets/individualcontribution.pdf).

**Comment by dajulia3** [comment]
@chibi03 closing this issue above since there is nothing more to say about it :)
